controller/src/main/java/io/pravega/controller/store/stream/PersistentStreamBase.java

```diff
@@ -471,88 +483,83 @@ private long findSegmentSplitsMerges(List<Segment> referenceSegmentsList, List<S
      * @param segmentsToSeal segments that will be sealed at the end of this scale operation.
      * @param newRanges      key ranges of new segments to be created
      * @param scaleTimestamp scaling timestamp
-     * @param runOnlyIfStarted run only if the scale operation was started. This is set to true only for manual scale.
+     * @param existing       existing epoch transition record
      * @return : list of newly created segments with current epoch
      */
     @Override
-    public CompletableFuture<EpochTransitionRecord> startScale(final List<Long> segmentsToSeal,
-                                                            final List<AbstractMap.SimpleEntry<Double, Double>> newRanges,
-                                                            final long scaleTimestamp,
-                                                            boolean runOnlyIfStarted) {
-        return verifyNotSealed().thenCompose(v -> getHistoryIndexFromStore()
-                .thenCompose(historyIndex -> getHistoryTableFromStore()
-                        .thenCompose(historyTable -> getSegmentIndexFromStore()
-                                .thenCompose(segmentIndex -> getSegmentTableFromStore()
-                                        .thenCompose(segmentTable -> {
-                                            if (!TableHelper.isScaleInputValid(segmentsToSeal, newRanges, segmentIndex.getData(),
-                                                    segmentTable.getData())) {
-                                                log.error("scale input invalid {} {}", segmentsToSeal, newRanges);
-                                                throw new EpochTransitionOperationExceptions.InputInvalidException();
-                                            }
+    public CompletableFuture<VersionedMetadata<EpochTransitionRecord>> submitScale(final List<Long> segmentsToSeal,
+                                                                                   final List<SimpleEntry<Double, Double>> newRanges,
+                                                                                   final long scaleTimestamp,
+                                                                                   final VersionedMetadata<EpochTransitionRecord> existing) {
+        return verifyNotSealed().thenCompose(v -> {
+            if (existing == null) {
+                return getEpochTransition();
+            } else {
+                return CompletableFuture.completedFuture(existing);
+            }
+        }).thenCompose(record -> {
+            if (!record.getObject().equals(EpochTransitionRecord.EMPTY)) {
+                // verify that its the same as the supplied input (--> segments to be sealed
+                // and new ranges are identical). else throw scale conflict exception
+                if (!verifyRecordMatchesInput(segmentsToSeal, newRanges, record.getObject())) {
+                    log.debug("scale conflict, another scale operation is ongoing");
```

Comment:
Actually whichever scaling is ongoing, it would be logged whenever it starts (if not started already). 
And that log is at info level. So we always know which scale started when and if there is conflict exception thrown, its unambiguous which scale it conflicted with. 
But we can add the additional information if needed.

Reply:
I think you are saying that this log line in combination with other scale request info in other log lines would provide enough info.

With respect to logging, I'm wondering if we should be adding the tracing here or in other log messages that we merged recently (PR #2803).

Reply:
The tracing is recorded in the logs in ScaleOperationTask (which is caller for these store methods).
As part of 2803, the `requestId` is not carried into the metadata store but used in level logging in the processors.